### ISSUE TITLE
problems met during inference

### ISSUE THREAD

**Issue by BaohaoLiao** [Yuanhy1997/SeqDiffuSeq#7]
Hi,

I could successfully train a model on IWSLT De-En. Now I have some questions about inference:

1. `decoder_attention_mask` is not in the args. Should I just comment it out? https://github.com/Yuanhy1997/SeqDiffuSeq/blob/ed56ca480002b89a26ebd16b0257e1aa880f80f1/inference_main.py#L124
2. How could I get the BLEU score after I run
`bash ./inference_scrpts/iwslt_inf.sh path-to-ckpts/ema_0.9999_280000.pt path-to-save-results path-to-ckpts/alpha_cumprod_step_260000.npy`
3. It seems you remove the code for DDIM. Do I have to keep the number of timesteps for inference the same as the one for training?

**Comment by Yuanhy1997** [comment]
1. We do not need the decoder attention mask arg. I should've delete that part of codes.
2. use the bleu_eval.py script for bleu.
3. Our code is not a method based on DDIM. You can explore to sample with DDIM.

**Comment by BaohaoLiao** [comment]
I can get the BLEU score now. Thank you very much.
